### ISSUE TITLE
Can't pass arguments to applications using "flutter run"

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#132002]
Desktop applications can have command line arguments passed to them. This works file when using "flutter build" and then calling the binary directly. However, when using "flutter run" there is no way to specify those command line arguments as far as I can tell. It would be great if there was a way to do this, like there is with "dart run".

**Comment by darshankawar** [comment]
> However, when using "flutter run" there is no way to specify those command line arguments as far as I can tell. It would be great if there was a way to do this, like there is with "dart run".



@Hixie Hi,

Can you please provide an example of the command with which we can't pass args currently as compared to `dart run` ?

**Comment by Hixie** [comment]
Yeah for example if your flutter app (called `foo`) has a `main.dart` that just says:

```dart
void main(List<String> arguments) => print(arguments);
```

...and you build and run it:

```bash
flutter build linux
build/linux/x64/release/bundle/foo a b c
```

...then it'll print `a b c`.

But there's no way to do this using `flutter run` as far as I can tell.

**Comment by christopherfujino** [comment]
This makes sense. I'm not sure if we would want to support `flutter run -d device a b c`, or require a new multi-option. For whoever implements this, there should probably be some design here.

**Comment by h3x4d3c1m4l** [comment]
Would it be possible to explain why this is closed as not planned @matanlurey? This is an actual nice feature to have...
